data_structures/linked_list/QueueLinkedList.c

```diff
@@ -0,0 +1,133 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+typedef struct queue
+{
+    int data;
+    struct queue *next;
+}queue;
+typedef struct Q
+{
+    queue *front;
+    queue *rear;
+}Q;
+void init(Q *p)
+{
+    p->front = p->rear = NULL;
+}
+int isEmpty(Q *p)
+{
+    if(p->front == NULL)
+        return 1;
+    return 0;
+}
+void enqueue(Q *p, int value)
+{
+    queue *q;
+    q = (queue*)malloc(sizeof(queue));
+    q->next = NULL;
+    q->data = value;
+    if(isEmpty(p))
+    {
+        p->front = p->rear = q;
+    }
+    else
+    {
+
+        (p->rear)->next = q;
+        p->rear = q;
+
+    }
+}
+int dequeue(Q *p)
+{
+    int x ;
+    queue *q;
+    q = p->front;
+    x = q->data;
+    if(q == p->rear)
+        init(p);
+    else
+        p->front = q->next;
+    free(q);
+    return x;
+}
+
+void display(Q *p)
+{
+    queue *q;
+    q = p->front;
+    while(q != NULL)
+    {
+        printf("%d\n", q->data);
+        q = q->next;
+    }
+}
+queue* search(Q *p, int item)
+{
+    queue *q;
+    q = p->front;
+    while(q != NULL)
+    {
+        if(q->data == item)
+            return q;
+        q = q->next;
+    }
+    printf("Element NOT Found.");
```

Comment:
The non-void function that you have written here doesn't always return a value in all control paths. Please fix this issue.